### PR TITLE
machineset with rhcos workers to scale to zero instead of delete

### PR DESCRIPTION
**Pull request by miyadav**
As @huali9  pointed out our failing cases were due to machineset not present at all  , PTAL when time permits if it's ok to keep them as default .

@jhou1 @gpei @sunzhaohua2 @jianli-wei

### CODE REVIEW

File: ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/disconnected/workers-rhel8/provision/cucushift-installer-rehearse-gcp-ipi-disconnected-workers-rhel8-provision-chain.yaml [hunk 1]
```diff
@@ -26,9 +26,9 @@ chain:
     default:  "8.6"
     documentation: "RHEL OS version"
   - name: REMOVE_RHCOS_WORKER
-    default: "yes"
+    default: "no"
```
Comment (jianli-wei) [inline ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/disconnected/workers-rhel8/provision/cucushift-installer-rehearse-gcp-ipi-disconnected-workers-rhel8-provision-chain.yaml:29]: Do you also want to update for other workflows? 
```
[jiwei@jiwei release]$ grep REMOVE_RHCOS_WORKER ci-operator/step-registry/cucushift/installer/rehearse/gcp/ -rn -A1
ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/disconnected/workers-rhel8/provision/cucushift-installer-rehearse-gcp-ipi-disconnected-workers-rhel8-provision-chain.yaml:28:  - name: REMOVE_RHCOS_WORKER
ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/disconnected/workers-rhel8/provision/cucushift-installer-rehearse-gcp-ipi-disconnected-workers-rhel8-provision-chain.yaml-29-    default: "yes"
--
ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/workers-rhel8/provision/cucushift-installer-rehearse-gcp-ipi-workers-rhel8-provision-chain.yaml:28:  - name: REMOVE_RHCOS_WORKER
ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/workers-rhel8/provision/cucushift-installer-rehearse-gcp-ipi-workers-rhel8-provision-chain.yaml-29-    default: "yes"
--
ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/workers-rhel8/fips/provision/cucushift-installer-rehearse-gcp-ipi-workers-rhel8-fips-provision-chain.yaml:29:  - name: REMOVE_RHCOS_WORKER
ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/workers-rhel8/fips/provision/cucushift-installer-rehearse-gcp-ipi-workers-rhel8-fips-provision-chain.yaml-30-    default: "yes"
[jiwei@jiwei release]$

**Comment by gpei** [comment]
@miyadav Hi, as we discussed in the Slack, we could consider keeping the RHCOS worker machineset later, but not updating REMOVE_RHCOS_WORKER to "no", this is related to what this workflow is going to test, and affecting our whole test coverage plan.

For this one `gcp-ipi-disconnected-workers-rhel8`, RHCOS worker should be removed.

**Comment by miyadav** [comment]
> @miyadav Hi, as we discussed in the Slack, we could consider keeping the RHCOS worker machineset later, but not updating REMOVE_RHCOS_WORKER to "no", this is related to what this workflow is going to test, and affecting our whole test coverage plan.
> 
> For this one `gcp-ipi-disconnected-workers-rhel8`, RHCOS worker should be removed.

Thanks @gpei , acked . Will prefer the other method so we will have better control where to enable using ENV variables.

**Comment by miyadav** [comment]
> > @miyadav Hi, as we discussed in the Slack, we could consider keeping the RHCOS worker machineset later, but not updating REMOVE_RHCOS_WORKER to "no", this is related to what this workflow is going to test, and affecting our whole test coverage plan.
> > For this one `gcp-ipi-disconnected-workers-rhel8`, RHCOS worker should be removed.
> 
> Thanks @gpei , acked . Will prefer the other method so we will have better control where to enable using ENV variables.



> @miyadav Hi, as we discussed in the Slack, we could consider keeping the RHCOS worker machineset later, but not updating REMOVE_RHCOS_WORKER to "no", this is related to what this workflow is going to test, and affecting our whole test coverage plan.
> 
> For this one `gcp-ipi-disconnected-workers-rhel8`, RHCOS worker should be removed.

@gpei , scaling to zero also removes them (just that machineset with '0' replicas can be queried in cluster) , the new machineset created by cloud team for testing would be removed as well after the scenario finishes , will that be ok ?

**Comment by gpei** [comment]
yes, this looks good to me, let's make a smoke test for this change

**Comment by gpei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-gcp-ipi-disconnected-workers-rhel8-p2-f7

**Comment by gpei** [comment]
In the rehearse job, we can see the worker machineset was set to "replica: 0" correctly:
```
+ echo '2023-03-06 05:06:52+00:00 - Deleting/scaling CoreOS machinesets to zero'
2023-03-06 05:06:52+00:00 - Deleting/scaling CoreOS machinesets to zero
+ mapfile -t COREOS_MACHINE_SETS
++ oc get machinesets --namespace openshift-machine-api
++ grep worker
++ grep -v rhel
++ awk '{print $1}'
+ [[ 4 != 0 ]]
+ oc scale machinesets --namespace openshift-machine-api ci-op-r97fiyds-35f0f-gtsp8-worker-a ci-op-r97fiyds-35f0f-gtsp8-worker-b ci-op-r97fiyds-35f0f-gtsp8-worker-c ci-op-r97fiyds-35f0f-gtsp8-worker-f --replicas 0
machineset.machine.openshift.io/ci-op-r97fiyds-35f0f-gtsp8-worker-a scaled
machineset.machine.openshift.io/ci-op-r97fiyds-35f0f-gtsp8-worker-b scaled
machineset.machine.openshift.io/ci-op-r97fiyds-35f0f-gtsp8-worker-c scaled
machineset.machine.openshift.io/ci-op-r97fiyds-35f0f-gtsp8-worker-f scaled
```

And only RHEL workers left in the cluster:
```
NAME                                                          STATUS                     ROLES                  AGE     VERSION           INTERNAL-IP   EXTERNAL-IP   OS-IMAGE                                                        KERNEL-VERSION                 CONTAINER-RUNTIME
ci-op-r97fiyds-35f0f-gtsp8-master-0.c.XXXXXXXXXXXX.internal   Ready                      control-plane,master   5h1m    v1.26.1+8cfbab7   10.0.0.5      <none>        Red Hat Enterprise Linux CoreOS 413.86.202303021410-0 (Ootpa)   4.18.0-372.43.1.el8_6.x86_64   cri-o://1.26.1-10.rhaos4.13.gitcb86088.el8
ci-op-r97fiyds-35f0f-gtsp8-master-1.c.XXXXXXXXXXXX.internal   Ready                      control-plane,master   5h2m    v1.26.1+8cfbab7   10.0.0.6      <none>        Red Hat Enterprise Linux CoreOS 413.86.202303021410-0 (Ootpa)   4.18.0-372.43.1.el8_6.x86_64   cri-o://1.26.1-10.rhaos4.13.gitcb86088.el8
ci-op-r97fiyds-35f0f-gtsp8-master-2.c.XXXXXXXXXXXX.internal   Ready                      control-plane,master   5h      v1.26.1+8cfbab7   10.0.0.4      <none>        Red Hat Enterprise Linux CoreOS 413.86.202303021410-0 (Ootpa)   4.18.0-372.43.1.el8_6.x86_64   cri-o://1.26.1-10.rhaos4.13.gitcb86088.el8
ci-op-r97fiyds-35f0f-gtsp8-rhel-1                             Ready                      worker                 4h18m   v1.26.1+8cfbab7   10.0.32.5     <none>        Red Hat Enterprise Linux 8.6 (Ootpa)                            4.18.0-425.13.1.el8_7.x86_64   cri-o://1.26.1-10.rhaos4.13.gitcb86088.el8
ci-op-r97fiyds-35f0f-gtsp8-rhel-2                             Ready,SchedulingDisabled   worker                 4h18m   v1.26.1+8cfbab7   10.0.32.6     <none>        Red Hat Enterprise Linux 8.6 (Ootpa)                            4.18.0-425.13.1.el8_7.x86_64   cri-o://1.26.1-10.rhaos4.13.gitcb86088.el8
```

**Comment by gpei** [comment]
@yunjiang29 @jianlinliu Could you help to approve this, Cloud team complains the worker machineset was deleted in such RHEL worker IPI cluster which caused many Cloud team's E2E cases failed, although I think deleting the RHCOS worker machineset is a reasonable action for such cluster which only have RHEL workers, but from Cloud team's perspective, by setting the replicas to "0" would be more friendly for their E2E debugging.

**Comment by jianlinliu** [comment]
> Cloud team complains the worker machineset was deleted in such RHEL worker IPI cluster which caused many Cloud team's E2E cases failed

So cloud team's e2e case will scale up rhcos worker again when we are testing against a specific cluster environment that we do not want any rhcos workers on purpose?

**Comment by jianlinliu** [comment]
> our machineset scaling related tests run serially and once tests are done , all resources , machineset ,autoscalers , workers will be removed . 

I guess we can not 100% guarantee  the created worker resources are always deleted, especially when some unexpected issues happened . So the best and safest thing is not to run these test cases in an improper configurations. So save testing time and cloud quota usage.

**Comment by miyadav** [comment]
> > our machineset scaling related tests run serially and once tests are done , all resources , machineset ,autoscalers , workers will be removed .
> 
> I guess we can not 100% guarantee the created worker resources are always deleted, especially when some unexpected issues happened . So the best and safest way is not to run these test cases in an improper configurations environment. Also save testing time and cloud quota usage.

Thanks @jianlinliu ,  After discussing internally with the team closing this one , the other PR - https://github.com/openshift/verification-tests/pull/3247 should help us avoid failures and unnecessary debug due to no RHCOS workers machines present .

**Pull request closed without merging**